crates/btcio/src/reader/state.rs

```diff
@@ -13,24 +14,52 @@ pub struct ReaderState {
 
     /// Depth at which we start pulling recent blocks out of the front of the queue.
     max_depth: usize,
+
+    /// Current transaction filtering config
+    filter_config: TxFilterConfig,
+
+    /// Current epoch
+    epoch: u64,
 }
 
 impl ReaderState {
     /// Constructs a new reader state instance using some context about how we
     /// want to manage it.
-    pub fn new(next_height: u64, max_depth: usize, recent_blocks: VecDeque<BlockHash>) -> Self {
+    // FIXME: the args list is getting long
```

Comment:
Can you remove this `FIXME`?
This will be handled by chippy's `too_many_arguments`: https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments
The default is 7 so we are good so far.